syntax = "proto3";
package pb;
option go_package = "pb";

message FileInfo{
    string name = 1;
    uint32 size = 2;
}

message UploadFileRequest{
    oneof data{
        FileInfo info = 1;
        bytes chunk_data = 2;
    }
}

message UploadFileResponse{
    string id = 1;
    uint32 size = 2;
}

service LaptopService{
    rpc UploadFile(stream UploadFileRequest) returns (UploadFileResponse) {};
}


//protoc --grpc_out=../pb --plugin=protoc-gen-grpc=`which grpc_cpp_plugin` file_service.proto 
//protoc --cpp_out=../pb file_service.proto